crates/miden-objects/src/block/header.rs

```diff
@@ -249,6 +266,213 @@ impl BlockHeader {
     }
 }
 
+impl TryFrom<ProposedBlock> for BlockHeader {
+    type Error = BlockHeaderError;
+
+    fn try_from(proposed_block: ProposedBlock) -> Result<Self, Self::Error> {
+        // Get the block number and timestamp of the new block and compute the tx commitment.
+        let block_num = proposed_block.block_num();
+        let timestamp = proposed_block.timestamp();
+
+        // Split the proposed block into its parts.
+        let (
+            batches,
+            account_updated_witnesses,
+            output_note_batches,
+            created_nullifiers,
+            partial_blockchain,
+            prev_block_header,
+        ) = proposed_block.into_parts();
+
+        let prev_block_commitment = prev_block_header.commitment();
+
+        // For now we copy the parameters of the previous header, which means the parameters set on
+        // the genesis block will be passed through. Eventually, the contained base fees will be
+        // updated based on the demand in the currently proposed block.
+        let fee_parameters = prev_block_header.fee_parameters().clone();
+
+        // Compute the root of the block note tree.
+        let note_tree = compute_block_note_tree(&output_note_batches);
+        let note_root = note_tree.root();
+
+        // Insert the created nullifiers into the nullifier tree to compute its new root.
+        let new_nullifier_root =
+            compute_nullifier_root(created_nullifiers, &prev_block_header, block_num)?;
+
+        // Insert the state commitments of updated accounts into the account tree to compute its new
+        // root.
+        let new_account_root =
+            compute_account_root(&account_updated_witnesses, &prev_block_header)?;
+
+        // Insert the previous block header into the block partial blockchain to get the new chain
+        // commitment.
+        let new_chain_commitment = compute_chain_commitment(partial_blockchain, prev_block_header);
+
+        // Aggregate the verified transactions of all batches.
+        let txs = batches.into_transactions();
+        let tx_commitment = txs.commitment();
+
+        // Construct the new block header.
+
+        // Currently undefined and reserved for future use.
+        // See miden-base/1155.
+        let version = 0;
+
+        // For now, we're not actually proving the block.
+        let proof_commitment = Word::empty();
+
+        // TODO: is this used? remove from block header?
+        let tx_kernel_commitment = Word::empty();
```

Comment:
@bobbinth will the block header need this field moving forward? I'm trying to keep the `ProposedBlock->BlockHeader` conversion in `miden-objects` but the `TransactionKernel` lives in miden-lib. I also have `ProposedBlock::sign() -> SignedBlock`, on that note.